packages/types/src/chain-adapters/index.ts

```diff
@@ -150,6 +151,11 @@ export type TradeDetails = {
   type: TradeType
 }
 
+export interface TxMetadata {
+  method?: string
+  parser?: string
```

Comment:
If parser metadata is included, should we at least require these two fields? I know this currently matches the payload coming from unchained, but just got me thinking.

Reply:
That's a fair point, though at the moment `method` is always undefined for zrx parsing, as we haven't yet sourced and added an ABI for it.

Reply:
True. Let's keep it in mind as a required base metadata type that we can extend if needed if it makes sense.

Reply:
Agree - I've made `parser` required for now in https://github.com/shapeshift/lib/pull/396/commits/b768e20811faf34b50379a38a189637b77e2336e.